js-intro.js

```diff
@@ -46,16 +99,68 @@ var stringWithVowels2 = "ILoveJavaScript"
 var notAString1 = true
 var notAString2 = 42
 
+//added extra variable with a string to test if function would catch it.
+
+var string1 ='HeyHowAreYou'
+
+const noStrAllowed = str => {
+
+    //checks if the input is a string or not
+
+    if(typeof str !== "string"){
+        return "This is not a string"
+    }
 
+    //create a new array that comes from input. Creates array with indivudal letters
 
+    var array = str.split('')
+
+    //create new array that takes input array value and filters out vowels.
+
+    var newArr1 = array.filter(value => value !== 'a' && value !== 'e' && value !== 'i' && value !== 'o' && value!== 'u')
+
+    //now that the vowels have been removed from the array, return the array into a string with no vowels.
+
+    return newArr1.join('')
+}
+
+//call all the variables.
+console.log(noStrAllowed(notAString1))
+console.log(noStrAllowed(notAString2))
+console.log(noStrAllowed(string1))
 
 // --------------------6) Create a function that takes the toonimals array and returns an array with only the toon objects that are cats.
 // Expected output: [ { name: "Stimpy", animal: "cat" }, { name: "Scratchy", animal: "cat" }, { name: "Felix", animal: "cat" } ]
 
 
 var toonimals = [ {name: "Itchy", animal: "mouse"}, {name: "Stimpy", animal: "cat"}, {name: "Daffy", animal: "duck"}, {name: "Scratchy", animal: "cat"}, {name: "Ren", animal: "dog"}, {name: "Felix", animal: "cat"}]
 
+//use filter to check for array values that contain "cat"
+var onlyCats = toonimals.filter(value => value.animal === "cat" )
 
+//log onlyCats value
+console.log(onlyCats)
 
 // --------------------7) Using the toonimals variable from #6, create a function that returns only the names of the non-cats.
 // Expected output: "Itchy" "Daffy" "Ren"
+
+
+//filtered out non cat values into new array
+
+var noCats = toonimals.filter(value => value.animal !== "cat" )
+
+//created function that pulls only object values relating to "names"
+
+const noCatNames = arr => {
```

Comment:
You're passing in `arr` as an argument but `arr` isn't used anywhere else in the function